pkg/providers/amifamily/resolver.go

```diff
@@ -48,11 +48,12 @@ type Resolver struct {
 
 // Options define the static launch template parameters
 type Options struct {
-	ClusterName             string
-	ClusterEndpoint         string
-	AWSENILimitedPodDensity bool
-	InstanceProfile         string
-	CABundle                *string `hash:"ignore"`
+	ClusterName              string
+	ClusterEndpoint          string
+	AWSENILimitedPodDensity  bool
+	InstanceProfile          string
+	CABundle                 *string `hash:"ignore"`
+	AssociatePublicIpAddress *bool   //nolint:revive,stylecheck
```

Comment:
Why is there a need for a `//nolint` here?

Reply:
Got it. I'm more inclined to side with the linter and not have `//nolint`s in the code if we can avoid it, even if the naming is slightly off